src/mrdsrg-helper/dsrg_mem.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
+ * that implements a variety of quantum chemistry methods for strongly
+ * correlated electrons.
+ *
+ * Copyright (c) 2012-2020 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ * @END LICENSE
+ */
+
+#ifndef _dsrg_mem_h_
+#define _dsrg_mem_h_
+
+#include <vector>
+#include <string>
+#include <map>
+
+#include "base_classes/mo_space_info.h"
+
+namespace forte {
+
+class DSRG_MEM {
```

Comment:
Can this be used for other codes? If it can then let's not be specific (DSRG). It might be nice to start it outside of the DSRG codes. However, this might not play well with python.

Reply:
This class is just for memory inspection before any actual DSRG computation. So this class might be useful for other dynamical correlation solver.